### PR TITLE
Abstract current AccessRoleProvider and implement as RbAclAccessRoles…

### PR DESCRIPTION
**Pull request by whikloj**
…Provider for use with WebAC.

Addresses [FCREPO-1716](https://jira.duraspace.org/browse/FCREPO-1716)

### CODE REVIEW

File: fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateRemoveChildrenRecursiveTest.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 import static org.mockito.Mockito.when;
 import static org.mockito.MockitoAnnotations.initMocks;
 
-import org.fcrepo.auth.roles.common.AccessRolesProvider;
+import org.fcrepo.auth.roles.common.RbAclAccessRolesProvider;
```
Comment (awoods) [inline fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateRemoveChildrenRecursiveTest.java]: Revert this change

File: fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateRemoveChildrenRecursiveTest.java [hunk 2]
```diff
@@ -53,7 +53,7 @@
     private BasicRolesAuthorizationDelegate authorizationDelegate;
 
     @Mock
-    private AccessRolesProvider accessRolesProvider;
+    private RbAclAccessRolesProvider accessRolesProvider;
```
Comment (awoods) [inline fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateRemoveChildrenRecursiveTest.java]: Revert this change

File: fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateTest.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
 import static org.mockito.Mockito.when;
 import static org.mockito.MockitoAnnotations.initMocks;
 
-import org.fcrepo.auth.roles.common.AccessRolesProvider;
+import org.fcrepo.auth.roles.common.RbAclAccessRolesProvider;
```
Comment (awoods) [inline fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateTest.java]: Revert this change

File: fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateTest.java [hunk 4]
```diff
@@ -55,7 +55,7 @@
     private BasicRolesAuthorizationDelegate authorizationDelegate;
 
     @Mock
-    private AccessRolesProvider accessRolesProvider;
+    private RbAclAccessRolesProvider accessRolesProvider;
```
Comment (awoods) [inline fcrepo-auth-roles-basic/src/test/java/org/fcrepo/auth/roles/basic/BasicRolesAuthorizationDelegateTest.java]: Revert this change

File: fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AbstractRolesAuthorizationDelegate.java [hunk 5]
```diff
@@ -50,7 +50,7 @@
     private static final String[] REMOVE_ACTIONS = {"remove"};
 
     @Autowired
-    private AccessRolesProvider accessRolesProvider = null;
+    private RbAclAccessRolesProvider accessRolesProvider = null;
```
Comment (awoods) [inline fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AbstractRolesAuthorizationDelegate.java]: Revert this change

File: fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRoles.java [hunk 6]
```diff
@@ -77,7 +77,7 @@
 
     @Inject
     @Optional
-    private AccessRolesProvider accessRolesProvider;
+    private RbAclAccessRolesProvider accessRolesProvider;
```
Comment (awoods) [inline fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRoles.java]: Revert this change

File: fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRoles.java [hunk 7]
```diff
@@ -108,7 +108,7 @@ public AccessRoles(final String externalPath) {
     /**
      * @return the accessRolesProvider
      */
-    private AccessRolesProvider getAccessRolesProvider() {
+    private RbAclAccessRolesProvider getAccessRolesProvider() {
```
Comment (awoods) [inline fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRoles.java]: Revert this change

File: fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRolesProvider.java [hunk 8]
```diff
@@ -15,146 +15,36 @@
...
-import static org.fcrepo.auth.roles.common.Constants.JcrName.assignment;
-import static org.fcrepo.auth.roles.common.Constants.JcrName.principal;
-import static org.fcrepo.auth.roles.common.Constants.JcrName.rbacl;
-import static org.fcrepo.auth.roles.common.Constants.JcrName.rbaclAssignable;
-import static org.fcrepo.auth.roles.common.Constants.JcrName.role;
-import static org.slf4j.LoggerFactory.getLogger;
 
 /**
- * Provides the effective access roles for authorization.
+ * Interface for the AccessRolesProvider
```
Comment (awoods) [inline fcrepo-auth-roles-common/src/main/java/org/fcrepo/auth/roles/common/AccessRolesProvider.java]: A "deeper" description of what this interface is designed for would be helpful.

File: fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesProviderTest.java [hunk 9]
```diff
@@ -78,14 +78,14 @@
 
     private NodeIterator rbaclIterator;
 
-    private AccessRolesProvider provider;
+    private RbAclAccessRolesProvider provider;
```
Comment (awoods) [inline fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesProviderTest.java]: Revert this change

File: fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesProviderTest.java [hunk 10]
```diff
@@ -78,14 +78,14 @@
...
 
     @Before
     public void setUp() throws RepositoryException {
         initMocks(this);
         when(node.getSession()).thenReturn(session);
 
-        provider = new AccessRolesProvider();
+        provider = new RbAclAccessRolesProvider();
```
Comment (awoods) [inline fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesProviderTest.java]: Revert this change

File: fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesTest.java [hunk 11]
```diff
@@ -59,7 +59,7 @@
 public class AccessRolesTest {
 
     @Mock
-    private AccessRolesProvider accessRolesProvider;
+    private RbAclAccessRolesProvider accessRolesProvider;
```
Comment (awoods) [inline fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesTest.java]: Revert this change

File: fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesProviderTest.java (hunk 10 above)
Comment (whikloj) [inline fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesProviderTest.java]: I get the others, but this would be trying to instantiate the interface. That can't work, no?

Comment (awoods) [inline fcrepo-auth-roles-common/src/test/java/org/fcrepo/auth/roles/common/AccessRolesProviderTest.java]: Correct. I got carried away.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo-module-auth-rbacl/commit/9e284e64cc982c9cea2e1fb116c9f736001fc429
